src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1389,6 +1389,39 @@ EnsureFunctionOwner(Oid functionId)
 }
 
 
+/*
+ * EnsureHashDistributedTable error out if the given relation is not a hash distributed table
+ * with the given message.
+ */
+void
+EnsureHashDistributedTable(Oid sourceRelationId)
+{
+	if (!isHashDistributedTable(sourceRelationId))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("Relation should be a "
```

Comment:
"Relation -> "relation \"%s\" ...", get_rel_name(sourceRelationId)

It would be nice to see the relation in question, since the user is entering two relations.

The postgres convention is for error messages to be lower case short messages rather than sentences (so no capital at the start, no dot at the end). We don't always follow that convention because users find it difficult to see the detail/hint messages in their application, but here we might as well.